buildenv/jenkins/API_check

```diff
@@ -0,0 +1,128 @@
+                // if there is an exception, set currentBuild.result to ABORTED/FAILURE
+                if (e.toString().contains("The job is completed without further action")){
+                    println("JCKUpdater script returned exit code 2, as no new update available. The job is completed without further action.")
+                    currentBuild.result = "SUCCESS"
+                } else if (e.toString().contains("FlowInterruptedException")) {
+                    currentBuild.result = 'ABORTED'
+                    println("Exception: " + e.toString())
+                } else {
+                    currentBuild.result = 'FAILURE'
+                    println("Exception: " + e.toString())
+                }
+            }
+        }
+    }
+}
+def forceCleanWS() {
+    try {
+        cleanWs disableDeferredWipeout: true, deleteDirs: true
+    } catch (Exception e) {
+        echo 'Exception: ' + e.toString()
+        //cleanWs has issue to delete workspace that contains non-ASCII filename in TKG output https://issues.jenkins.io/browse/JENKINS-33478
+        //cannot delete workspace directly. Otherwise, Jenkins job will abort due to missing workspace
+        sh "rm -rf ${env.WORKSPACE}/aqa-tests/TKG"
+        // call cleanWs() again
+        cleanWs disableDeferredWipeout: true, deleteDirs: true
+    }
+}
+
+def automateAPICheck() {
+    echo ("checking if new JCK material is available .....")
+    def ARTIFACTORY_API_URL_OPTION = params.ARTIFACTORY_API_URL ? "-au ${params.ARTIFACTORY_API_URL}": ""
+    stage('API_CHECK') {
+        withCredentials([
+            usernamePassword(credentialsId: "${params.ARTIFACTORY_CREDENTIALS}",
+			usernameVariable: 'ARTIFACTORY_USER', passwordVariable: 'ARTIFACTORY_CREDENTIALS')
+        ]){
+            // def currentDirectory = sh(script: 'pwd', returnStdout: true).trim()
+            dir("${WORKSPACE}/aqa-tests/jck/jck-semiauto-updater"){
+                def SCRIPT ="./update-check.sh ${ARTIFACTORY_API_URL_OPTION} -at \"\$ARTIFACTORY_CREDENTIALS\" "
+                def tempOutputFile = "temp_output.txt"
+                def EXITCODE = sh(script: "${SCRIPT} > ${tempOutputFile}",  returnStatus: true)
+
+                if (EXITCODE == 0) {
+                    def scriptOutput = readFile(tempOutputFile).trim()
+                    def jdkList = [:]
+                    testJobs = [:]
+                    echo ("Automate API script output:")
+                    echo scriptOutput
+
+                    jdkList= getJDKList(scriptOutput)
+                    jdkList.each { key, value ->
+                        println("JDK number: $key")
+                        def CHILD_JOB_NAME="JCK_Sync"
+                        def BUILD_URL="${JENKINS_URL}job/${CHILD_JOB_NAME}/"
+                        def CHILD_PARAMS = []
+                        def JCK_GIT_REPO = "git@github.ibm.com:runtimes/JCK${key}-unzipped.git"
```

Comment:
The script is also used by other vendors (e.g. Adoptium), not only ibm, so the link repo cannot be hardcoded.  Suggest a root location be passed into the script (jenkins env var) and JCK_GIT_REPO be constructed.